domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs

```diff
@@ -132,43 +133,43 @@ public async Task<List<Measurement>> FetchMeasurements(MeteringPointTimeSeriesSl
 
         if (dateFrom < synchronizationPointSeconds)
         {
-            try
+            var request = new GetMeasurementsRequest
             {
-                var request = new GetMeasurementsRequest
-                {
-                    DateFrom = dateFrom,
-                    DateTo = synchronizationPointSeconds,
-                    Gsrn = slidingWindow.GSRN,
-                    Subject = meteringPointOwner,
-                    Actor = Guid.NewGuid().ToString()
-                };
-                var res = await measurementsClient.GetMeasurementsAsync(request, cancellationToken: cancellationToken);
-
-                logger.LogInformation(
-                    "Successfully fetched {numberOfMeasurements} measurements for GSRN {GSRN} in period from {from} to: {to}",
-                    res.Measurements.Count,
-                    slidingWindow.GSRN,
-                    DateTimeOffset.FromUnixTimeSeconds(dateFrom).ToString("o"),
-                    DateTimeOffset.FromUnixTimeSeconds(synchronizationPointSeconds).ToString("o"));
-
-                return res.Measurements.ToList();
-            }
-            catch (Exception e)
-            {
-                logger.LogError("An error occured: {error}, no measurements were fetched", e.Message);
-            }
+                DateFrom = dateFrom,
+                DateTo = synchronizationPointSeconds,
+                Gsrn = slidingWindow.GSRN,
+                Subject = meteringPointOwner,
+                Actor = Guid.NewGuid().ToString()
+            };
+            var res = await measurementsClient.GetMeasurementsAsync(request, cancellationToken: cancellationToken);
+
+            logger.LogInformation(
+                "Successfully fetched {numberOfMeasurements} measurements for GSRN {GSRN} in period from {from} to: {to}",
+                res.Measurements.Count,
+                slidingWindow.GSRN,
+                DateTimeOffset.FromUnixTimeSeconds(dateFrom).ToString("o"),
+                DateTimeOffset.FromUnixTimeSeconds(synchronizationPointSeconds).ToString("o"));
+
+            return res.Measurements.ToList();
         }
 
         return new();
     }
 
     public async Task HandleSingleSyncInfo(MeteringPointSyncInfo syncInfo, CancellationToken stoppingToken)
     {
-        var slidingWindow = await slidingWindowState.GetSlidingWindowStartTime(syncInfo, stoppingToken);
+        try
```

Comment:
Du smækkede denne logik i en try block. Jeg er nysgerrig på hvorfor, og hvad det giver os. Kan du fortælle mere?

Reply:
Jeg har flyttet den try/catch konstruktion fra FetchMeasurements funktionen til HandleSingleSyncInfo funktionen, og det er for at fange exceptions som bliver smidt uden FetchMeasurements funktionen.